### ISSUE TITLE
(v2.0) exposed a bug moving class between symbol dicts by loading different config without default symbol dictionary

### ISSUE THREAD

**Issue by dalehenrich** [GemTalk/Rowan#493]
The test `RwRowanSample4Test debug: #testLoadProjectNamed_221A_v20_1` exposed the bug ... need to create an explicit/simpler test case for this bug and then add defaultSymbolDictionary to Core config so `RwRowanSample4Test debug: #testLoadProjectNamed_221A_v20_1` will pass again...

**Comment by dalehenrich** [comment]
... looks like we need more tests for this issue ... currently only two passing tests

**Comment by dalehenrich** [comment]
Here's my analysis for RwRowanSample4Test>>testLoadProjectNamed_493:
```
As simple as it sounds ... defaultSymbolDictName not defined in Issue_493 ... otherwise the same ... note that no groups are specified on load line in current test after loading issue_493:

rogue:components>diff Issue_493.ston Core.ston 
2c2
< 	#name : 'Issue_493',
---
> 	#name : 'Core',
39a40
> 				#defaultSymbolDictName : 'RowanSample4SymbolDict',

============================================================
The comparison in this case shows up with a gs_symbolDictionary modification from current symdict to _nil_ ...
============================================================
moved packages:
	Rowan-Sample4-Extensions
	Rowan-Sample4-Core
classes with property changes:						------ class property changes are pretty benign ... since it _is_ just a property change
	RowanSample4
	RowanSample4IdentityKeyValueDictionary	<-- what is the property change?
class with symbol dictionary changes:			------ this is the change that does the real work ... 
	RowanSample4
=============================================================
okay, RwGsClassSymbolDictionaryMoveSymDictPatch>>installSymbolDictionaryPatchFor: 
deletes the class from the old package, but does not ensure that the methods are 
moved to the new package! ...
```
In other move class scenarios, the methods are being indepently moved, but in this instance (effectively a changed defaultSymbolDict) the methods are not independently moved ...

**Comment by dalehenrich** [comment]
The fix for this bug is a little deeper than I wan to tackle right now ... the current test case, involves a change to the gs_symbolDictionary property ...the root problem is that class is being moved, but the loaded methods associated with the class are not handled correctly ... the fix will need to include the scheduling of the move of methods to the new symbol dictionary ... something that is done by moveClass and moveMethod ... but this particluar case is not recognized by the 'move detector" logic ... also I need a more complicated test case to cover this properly ...

**Comment by dalehenrich** [comment]
RwGsSymbolDictionaryRegistry_Implementation class>>ensureExtensionClassNamed:existsForPackageNamed:instance: gets called for the same extension (even though it is a move of the method), but the old extension method is not removed ... probably `RwGsPatchSet_254>>updateMethodDictionaries` or  `RwGsPatchSet_254>>doMoveMethodsBetweenPackages` needs to remove the old extension method registration ...  `doMoveMethodsBetweenPackages` is called **before** `updateMethodDictionaries` ... and we do *know* that the move is an extension method ...

**Comment by dalehenrich** [comment]
`RwGsPatchSet_254>>installAddedClassExtensions` (where the extra class extension is added) is called **before** `RwGsPatchSet_254>>doMoveMethodsBetweenPackages`, so we'll have to be sure to remove the correct extension method ...

**Comment by dalehenrich** [comment]
`RwGsPatchSet_254>>doMoveMethodsBetweenPackages` is attempting to remove the old loaded extension method from the the *after* registry, when it should be remove from the *before* registry.

**Comment by dalehenrich** [comment]
...  This bug has been fixed in v2.0 and the fix was specifically fixed by the creation of the two classes RwLoadSpecificationV2 and RwProjectSpecificationV2 and the proper handling of the default symbol dict name ... as this bug was tracked down to the fact that the wrong symbol dictionary name was being supplied by the Component and V1.2 implementations ...
